### DISCUSSION TITLE
Traceback (most recent call last): File "C:\Users\zzc\Desktop\practice\Smiles_to_Graph\smiles_to_struture.py", line 33, in <module> generation_images(df) File "C:\Users\zzc\Desktop\practice\Smiles_to_Graph\smiles_to_struture.py", line 18, in generation_images Draw.MolToFile(mol, f"./generation/img{i}.png", size=(150, 100)) File "C:\py\envs\pytorch\lib\site-packages\rdkit\Chem\Draw\__init__.py", line 312, in MolToFile with open(filename, 'w+' + mode) as outf: OSError: [Errno 22] Invalid argument: './generation/img*C*.png'

### DISCUSSION THREAD

**Discussion by ZhenchangZhang** [rdkit/rdkit#7622] [FAQ]
RDKit Version: '2024.03.1'
Operating system: windows 10, 64bits
Python version (if relevant): 3.9
Are you using conda? YES
If you are using conda, which channel did you install the rdkit from?
If you are not using conda: how did you install the RDKit?

import pandas as pd
import xlsxwriter
import openpyxl
from rdkit import Chem
from rdkit.Chem import Draw
from rdkit.Chem.Draw.MolDrawing import MolDrawing, DrawingOptions

# 读入要处理的数据
df = pd.read_csv('test.csv', )

# 根据smiles编码生成图片，生成的图片挡在generation文件夹中
def generation_images(data):
    draw = data
    for i in draw:
        mol = Chem.MolFromSmiles(i)
        Draw.MolToFile(mol, /generation/img.png", size=(150, 100))

# 创建excel表格，将生成的图片读入
def load_images(data):
    workbook = xlsxwriter.Workbook('test.csv_dataset_with_iamges.xlsx')
    worksheet = workbook.add_worksheet()
    for i, j in enumerate(data):
        worksheet.write(f'A{i + 1}', f'{j}')
        worksheet.insert_image(f'B{i + 1}', f'./generation/img{j}.png')
    workbook.close()

generation_images(df)
load_images(df)

import pandas as pd
import xlsxwriter
import openpyxl

from rdkit import Chem
from rdkit.Chem import Draw
from rdkit.Chem.Draw.MolDrawing import MolDrawing, DrawingOptions

# 读入要处理的数据
df = pd.read_csv('test.csv', )

# 根据smiles编码生成图片，生成的图片挡在generation文件夹中
def generation_images(data):
    draw = data
    for i in draw:
        mol = Chem.MolFromSmiles(i)
        Draw.MolToFile(mol, f"./generation/img{i}.png", size=(150, 100))

# 创建excel表格，将生成的图片读入
def load_images(data):
    workbook = xlsxwriter.Workbook('test.csv_dataset_with_iamges.xlsx')
    worksheet = workbook.add_worksheet()
    for i, j in enumerate(data):
        worksheet.write(f'A{i + 1}', f'{j}')
        worksheet.insert_image(f'B{i + 1}', f'./generation/img{j}.png')
    workbook.close()

generation_images(df)
load_images(df)

why  "Draw.MolToFile(mol, /generation/img.png", size=(150, 100))" succeed
but " Draw.MolToFile(mol, f"./generation/img{i}.png", size=(150, 100))" fail

Traceback (most recent call last):   File "C:\Users\zzc\Desktop\practice\Smiles_to_Graph\smiles_to_struture.py", line 33, in <module>     generation_images(df)   File "C:\Users\zzc\Desktop\practice\Smiles_to_Graph\smiles_to_struture.py", line 18, in generation_images     Draw.MolToFile(mol, f"./generation/img{i}.png", size=(150, 100))   File "C:\py\envs\pytorch\lib\site-packages\rdkit\Chem\Draw\__init__.py", line 312, in MolToFile     with open(filename, 'w+' + mode) as outf: OSError: [Errno 22] Invalid argument: './generation/img*C*.png'

**Comment by greglandrum** [comment]
I'm closing this because it's so badly formatted that I can't even figure out what the question is.
@ZhenchangZhang if you want to try reformatting this so that it's clear what is code (and the code is properly formatted, read the help for this: https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax), feel free to do so and @ mention me and I will re-open it
